src/Chapter2.hs

```diff
@@ -842,7 +867,12 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate _ [] = []
+rotate n l
+    | n < 0 = []
+    | null l = []
+    | otherwise = take (length l) (drop n (cycle l))
```

Comment:
As another optimisation, you can use `mod` by the length of the list on `n`, so you won't do unnecessary work 🙂